src/btree/bt_cursor.c

```diff
@@ -638,6 +640,7 @@ __wt_btcur_search_near(WT_CURSOR_BTREE *cbt, int *exactp)
      * they currently have pinned.)
      */
     WT_ERR(__cursor_localkey(cursor));
+    __cursor_free_value(session, cursor);
```

Comment:
I would like to use different names than `__cursor_unset_xxx` and `__cursor_free_xxx`, because we're not only "unsetting" or "freeing" things, we're switching to a different chunk of memory, so I think it's confusing.